File: src/javascript/user/message/delete-message.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

if (process.env.FAXCORE_CONFIRM_DESTRUCTIVE !== 'true') {
  throw new Error('Set FAXCORE_CONFIRM_DESTRUCTIVE=true to delete messages.');
}

const messageID = process.env.FAXCORE_MESSAGE_ID;
if (!messageID) {
  throw new Error('FAXCORE_MESSAGE_ID is required.');
}

const client = new FaxCoreClient();
const result = await client.delete('/api/message/delete', {
  messageID: messageID.split(',').map((value) => value.trim()).filter(Boolean)
});

printResult('Delete message', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/Message/DeleteMessage.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Message
{
    internal static class DeleteMessage
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            RequireDestructiveConfirmation();
            if (string.IsNullOrWhiteSpace(config.MessageId))
            {
                throw new InvalidOperationException("Set FAXCORE_MESSAGE_ID before running user.message.delete.");
            }

            return client.DeleteAsync("/api/message/delete", new { messageID = SplitCsv(config.MessageId) });
        }

        private static string[] SplitCsv(string value)
        {
            return value.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
        }

        private static void RequireDestructiveConfirmation()
        {
            if (!string.Equals(Environment.GetEnvironmentVariable("FAXCORE_CONFIRM_DESTRUCTIVE"), "true", StringComparison.OrdinalIgnoreCase))
            {
                throw new InvalidOperationException("Set FAXCORE_CONFIRM_DESTRUCTIVE=true to delete messages.");
            }
        }
    }
}

